samples/dispatch-es6/app.js

```diff
@@ -0,0 +1,182 @@
+const { BotFrameworkAdapter, ConversationState, MemoryStorage, TurnContext } = require('botbuilder');
+const { LuisRecognizer, QnAMaker } = require('botbuilder-ai');
+const { DialogSet } = require('botbuilder-dialogs');
+const restify = require('restify');
+
+// Create server
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`${server.name} listening to ${server.url}`);
+});
+
+// Create adapter
+const adapter = new BotFrameworkAdapter({
+    appId: process.env.MICROSOFT_APP_ID,
+    appPassword: process.env.MICROSOFT_APP_PASSWORD
+});
+
+// Create LuisRecognizers and QnAMaker
+// The LUIS applications are public, meaning you can use your own subscription key to test the applications.
+// For QnAMaker, users are required to create their own knowledge base.
+// The exported LUIS applications and QnAMaker knowledge base can be found adjacent to this sample bot.
+
+// The corresponding LUIS application JSON is `dispatchSample.json`
+const dispatcher = new LuisRecognizer({
+    appId: '0b18ab4f-5c3d-4724-8b0b-191015b48ea9',
+    subscriptionKey: process.env.LUIS_SUBSCRIPTION_KEY,
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com/',
+    verbose: true
+});
+
+// The corresponding LUIS application JSON is `homeautomation.json`
+const homeAutomation = new LuisRecognizer({
+    appId: 'c6d161a5-e3e5-4982-8726-3ecec9b4ed8d',
+    subscriptionKey: process.env.LUIS_SUBSCRIPTION_KEY,
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com/',
+    verbose: true
+});
+
+// The corresponding LUIS application JSON is `weather.json`
+const weather = new LuisRecognizer({
+    appId: '9d0c9e9d-ce04-4257-a08a-a612955f2fb5',
+    subscriptionKey: process.env.LUIS_SUBSCRIPTION_KEY,
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com/',
+    verbose: true
+});
+
+// The QnAMaker knowledge base used in this sample is `sampleKnowledgeBase.tsv`
+// To create a new service, upload this knowledge base by going to QnA Maker at https://qnamaker.ai/Create
+const faq = new QnAMaker({
+    knowledgeBaseId: '',
+    subscriptionKey: '',
+    serviceEndpoint: 'https://westus.api.cognitive.microsoft.com/'
+});
+
+// create conversation state
+const conversationState = new ConversationState(new MemoryStorage());
+adapter.use(conversationState);
```

Comment:
Cash and I have decided that we should always show the use of both UserState() and ConversationState() in all samples with state. I can send you the general boilerplate to include